### PR TITLE
Configurabiity of the UDP port # for messaging to statsd

### PR DESCRIPTION
**Pull request by ravisinghsfbay**
Testing done:
1. Default port # gets setup correctly
2. Configured value takes effect
3. Changing the configured value takes effect
4. Allow admin to choose any UDP port, while being mindful of what is usable.

Note:
1. Port # will be a string in the Json config configured as below in /etc/glb/director.conf
	"statsd_port": "12345"
2. For a config change to take effect, the glb-director will need to be restarted. This is true
   of all configurable fields in /etc/glb/director.conf

### CODE REVIEW

File: src/glb-director/glb_director_config.c [hunk 1]
```diff
@@ -107,6 +108,15 @@ glb_director_config *glb_director_config_load_file(const char *config_file,
 	json_t *item;
 	int ret;
 
+	item = json_object_get(root, "statsd_port");
+	if (item != NULL && json_is_string(item)) {
+		uint16_t port;
+		port = atoi(json_string_value(item));
```
Comment (theojulienne) [inline src/glb-director/glb_director_config.c]:
```suggestion
		port = json_integer_value(item);
```

**Comment by theojulienne** [comment]
Would also be great to add the new field to the [config file documentation](https://github.com/github/glb-director/blob/master/docs/setup/glb-director-configuration.md#etcglbdirectorconf) as part of this PR too, just to keep it in sync 🙇

**Comment by ravisinghsfbay** [comment]
Thanks for the comments. 
All have been addressed.

File: docs/setup/glb-director-configuration.md [hunk 2]
```diff
@@ -138,3 +138,10 @@ A distributor lcore performs at minimum `rx` work (receiving from a NIC queue),
 Worker lcores take packets received by the distributor, hash them, encapsulate them, then send them back to the distributor to be sent to the NIC. For more details of the technical implementation of this and why it's used, see [DPDK Packet Distributor Library](https://doc.dpdk.org/guides/prog_guide/packet_distrib_lib.html).
 
 `work_source` configures the lcore index of the distributor core that work will be sourced from. The number of cores that reference any given distributor should match that distributor core's `num_dist_workers`.
+
+### `statsd_port`
+
+_optional_
+
+By default, GLB Director uses UDP port 28125 to send messages to statsd.
+This configuration option, makes the port number configurable.
```
Comment (theojulienne) [inline docs/setup/glb-director-configuration.md]:
```suggestion
This configuration option allows selecting a different port as the target for statsd messages.
```

**Pull request merged**